src/Adapter/Customer/CommandHandler/AbstractCustomerHandler.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Customer\CommandHandler;
+
+use Customer;
+use PrestaShop\PrestaShop\Core\Domain\Customer\Exception\CustomerNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Customer\ValueObject\CustomerId;
+
+abstract class AbstractCustomerHandler
+{
+    /**
+     * @param CustomerId $customerId
+     * @param Customer $customer
+     *
+     * @throws CustomerNotFoundException
+     */
+    protected function assertCustomerWasFound(CustomerId $customerId, Customer $customer)
+    {
+        if (!$customer->id) {
```

Comment:
shouldn't be this condition instead? I don't understand the error message linked to the failed assertion:

```php

if ($customer->id !== $customer->getValue())
```

Reply:
i guess it could be done that way as well, wont change the behavior.

Reply:
I think it's more readable, go for it please :)

Reply:
and done. :+1: